Support VCS path variance

squidadm opened the pull request:
Stop hard coding the VCS checkout location.
Instead use a configuration file as done for the website path.

squidadm committed:
Support VCS path variance

Stop hard coding the VCS checkout location and use a
configuration file as done for the website path.

rousskov commented on release/mk-langpack.sh:
```diff
@@ -13,7 +13,13 @@ webpath=`cat ~/.website-path`
+vcspath=`cat ~/.code-vcs-path`
+vcsver=`ls -1 $vcspath | cut -d- -f2 | sort -h`
+vcsdir="${vcspath}squid-${vcsver}"
+if ! test -d $vcsdir ; then
```
I do not know enough to request specific improvements (or even to know where those improvements should be made), but please note that this change is inconsistent with `release/mk-release-series.sh` being imported in #8. The latter code calls the contents of the `~/.code-vcs-path` file "vcsdir" rather than "vcspath" and has no vcspath-to-vcsdir adjustments present in this PR above:

```sh
# Location of the repository checkouts
vcsdir=`cat ~/.code-vcs-path`
```

In other words, vcsdir here is different from vcsdir in #8 despite the same name and shared association with `~/.code-vcs-path`.

kinkie replied:
I think this change request has merit, we should be consistent if we define this internal API

yadij replied:
Done as of [4b0aaab](https://github.com/squid-cache/ci/pull/7/commits/4b0aaab12a3e24000471ab2a5030ba3e1452b87f)

squidadm committed:
Improve server environment configuration

see https://github.com/squid-cache/ci/pull/8#discussion_r1091755049

yadij commented on release/.server.config:
```diff
@@ -0,0 +1,14 @@
+#
+# Template configuration file for server environment
```
Note this is a template. Paths listed are for documentation purpses only.

squidadm committed:
Fix config template

Dash shell does weird things if the paths are not quoted.

squidadm committed:
Human sorting places the latest version last

rousskov commented on release/.server.config:
```diff
@@ -0,0 +1,17 @@
+SQUID_RSYNC_PATH="/var/rsync"
+
+# Squid build server
+SQUID_CI_BUILD="example.com.invalid"
```
CI is too general IMO -- we have at least three CI servers already. I do not know what "build server" is in this context, so I cannot suggest specific changes, but please _detail_ that in the comment/description and then rename the variable (e.g., to `SQUID_BUILD_SERVER` or something more suitable).

rousskov commented on release/mk-langpack.sh:
```diff
@@ -8,12 +8,20 @@
-# Location of the repository checkout
-vcsdir=/srv/sources/squid-master
+# Location of the latest versions repository checkout
+vcsdir="${SQUID_VCS_PATH}squid-${SQUID_RELEASE}"
```
This line contradicts `SQUID_VCS_PATH` definition (and the provided example) which places versions in subdirectories:

```suggestion
vcsdir="${SQUID_VCS_PATH}/squid-${SQUID_RELEASE}"
```

squidadm committed:
Fix typo in langpack script

squidadm committed:
Reviewer requested change

rousskov commented on release/.server.config:
```diff
@@ -0,0 +1,17 @@
+SQUID_RSYNC_PATH="/var/rsync"
+
+# Hostname of server performing CI tasks
+SQUID_CI_SERVER="example.com.invalid"
```
CI is too general IMO -- we have at least three CI servers already, all performing "CI tasks" (naturally). I do not know which "CI server" this variable defines, so I cannot suggest specific changes, but please detail that in the comment/description and then rename the variable (e.g., to SQUID_BUILD_SERVER or something more suitable).

yadij replied:
I am not sure either. It is not the name of a host doing the build. It is the name of the CI server coordinating the CI activities and storing the results of those. Thus the name is has now.

I would like to shelve this change request as "good enough" so I can move forward with importing the rest of the automation into this VCS. Followup improvements are welcome of course.

rousskov replied:
> It is the name of the CI server coordinating the CI activities and storing the results of those.

Adding that information to the comment would be slightly better than doing nothing, but I recommend to at least mention an example of a "CI activity" and an example of a "result". Otherwise, there are still too many CI servers that match the description. I know of several Squid CI servers, and I still do not know which one this variable is supposed to name. In fact, every one of them matches the description, for some reasonable definitions of "coordinating the CI activities" and "storing the results of CI activities".

AFAICT, the official squid-ci repository does not use this variable, and the PR itself does not use this variable, so I cannot easily guess what it is based on use cases either.

Going forward:

* If this problematic variable is currently unused in the official repository and this PR, and you want to move forward without fixing its problems, then do not add this variable in this PR.

* If you want to fix its problems, then consider sharing a specific example of this variable (or equivalent) use, current or future, and I will try to suggest a specific fix. The more examples you provide, the higher are the chances that my proposed fix will apply to all of them, of course.

yadij replied:
Dropped for now.

yadij replied:
Dropped this for now in [348f28b](https://github.com/squid-cache/ci/pull/7/commits/348f28b49b493d1005d121a6a4e51259533239d6)

rousskov committed:
Significantly improved to:

 - provide a useful error message instead of failing in complete silence
   when the configuration file is missing
 - simplify
 - avoid misleading a reader with what looks like "load an optional
   configuration file if it exists" logic but is actually "load a required
   configuration file" logic
 - remove code duplication

yadij committed:
Merge branch 'main' into flexible-vcs-path

squidadm committed:
Fix mk-cfgman-docs.sh permissions

squidadm committed:
Remove disputed variable

yadij committed:
Merge branch 'main' into flexible-vcs-path

rousskov reviewed (commented):
Thank you for addressing my concern.

squidadm committed:
Document config file

kinkie reviewed (approved):
Good for me, cleared to go
